src/networking/apiparser.cpp

```diff
@@ -0,0 +1,223 @@
+                                    discoveredServerAttributes.m_name;
+                         });
+    if (attribPtr != m_discoveredTcpServersAttrib.cend())
+    {
+        // Server with a given name has already been discovered
+        return;
+    }
+    // Finally, save information about server & emit "user discovered" signal
+    m_discoveredTcpServersAttrib.push_back(discoveredServerAttributes);
+
+    emit newUserDiscovered();
+}
+// ==========================================================================================
+// ==========================================================================================
+// ==========================================================================================
+//                                                                       GET DISCOVERED USERS
+QVector<QString> DefaultLocalConnector::getOnlineUsers() const
+{
+    QVector<QString> userNames;
+    // Get user names from discovered servers
+    for (const auto & serverAttrib : m_discoveredTcpServersAttrib)
+        userNames.push_back(serverAttrib.m_name);
+
+    return userNames;
+}
+// ==========================================================================================
+// ==========================================================================================
+// ==========================================================================================
+//                                                        GET USERS, ALLOWED FOR SHARING WITH
+QVector<QString> DefaultLocalConnector::getConnectedUsers() const
+{
+    QVector<QString> userNames;
+    // Get user names from connected servers
+    for (const auto & serverAttrib : m_connectedTcpServersAttrib)
+        userNames.push_back(serverAttrib.m_name);
+
+    return userNames;
+}
+// ==========================================================================================
+// ==========================================================================================
+// ==========================================================================================
+//                                                                            REQUEST SHARING
+void DefaultLocalConnector::requestSharingWithUser(QString userName)
+{
+    auto serverAttributesPtr =
+            std::find_if(m_discoveredTcpServersAttrib.begin(),
+                         m_discoveredTcpServersAttrib.end(),
+                         [userName] (const ServerData & serverData)
+                         {
+                             return serverData.m_name == userName;
+                         });
+    // If found user with a given userName
+    if (serverAttributesPtr != m_discoveredTcpServersAttrib.end())
+    {
+        m_connectedTcpServersAttrib.push_back(*serverAttributesPtr);
+        //m_tcpService->connectToTcpServer(*serverAttributes);
+    }
+}
+// ==========================================================================================
```

Comment:
All team should follow the same style. If you add these comments to each function, all team should follow the same rules.

Reply:
You can use http://www.doxygen.nl/ then generate documentation. This must be agreed by the team